library/common/http/client.cc

```diff
@@ -80,6 +80,16 @@ void Client::DirectStreamCallbacks::encodeData(Buffer::Instance& data, bool end_
     closeStream();
   }
 
+  if (data.length() != 0 && explicit_flow_control_ && !response_data_) {
```

Comment:
can we comment why this move is necessary? Sadly I don't think I grok the race condition here